apps/test-suite/tests/Location.js

```diff
@@ -17,6 +17,35 @@ export async function test(t) {
   const shouldSkipTestsRequiringPermissions = await TestUtils.shouldSkipTestsRequiringPermissionsAsync();
   const describeWithPermissions = shouldSkipTestsRequiringPermissions ? t.xdescribe : t.describe;
 
+  const testShapeOrUnauthorized = (testFunction, options) => async () => {
```

Comment:
I think we shouldn't pass `testFunction` and `options` separately, that way we're losing invocation context. It might be better to have just one argument here that is an arrow function that calls appropriate method with appropriate arguments.